encrypt-main.jsx

```diff
@@ -19,7 +19,7 @@ function redirect(state, replace) {
   if (state.location.query.video) {
     replace(`/encrypt/${pageType}/${state.location.query.video}`);
   } else {
-  	replace(`/encrypt/${pageType}/${encryptVideos.length}`);
+  	replace(`/encrypt/${pageType}/${encryptVideos.length}${state.location.search}`);
```

Comment:
You're saying use `replace` so that it doesn't add a second location to the browser history? Good idea.